assessment_store/db/queries/assessment_records/_helpers.py

```diff
@@ -10,147 +10,189 @@
-            "country": postcode_data["country"],
-            "constituency": postcode_data["parliamentary_constituency"],
-        }
-        if postcode_data
-        else None
-    )
-    return location_data
+    try:
+        response = requests.post(
+            url="http://api.postcodes.io/postcodes",
+            json={"postcodes": [raw_postcode]},
+            headers={"Content-Type": "application/json"},
+        )
+        if response.status_code != 200:
+            current_app.logger.error(
+                "Failed requesting location data for postcode {postcode}. Received error response {response}",
+                extra=dict(postcode=raw_postcode, response=response),
+            )
+            return None
+
+        result = response.json()
+        postcode_data = result["result"][0]["result"]
+        location_data = (
+            {
+                "error": False,
+                "postcode": raw_postcode,
+                "county": postcode_data["admin_county"]
+                if postcode_data["admin_county"]
+                else postcode_data["admin_district"],
+                "region": postcode_data["region"]
+                if postcode_data["region"]
+                else postcode_data["european_electoral_region"],
+                "country": postcode_data["country"],
+                "constituency": postcode_data["parliamentary_constituency"],
+            }
+            if postcode_data
+            else None
+        )
+        return location_data
+
+    except Exception as e:
+        current_app.logger.error(
+            "Failed requesting location data for postcode {postcode}", exc_info=e, extra=dict(postcode=raw_postcode)
+        )
+        return None
 
 
 def derive_application_values(application_json):  # noqa: C901 - historical sadness
     # TODO: implement mapping function to match
     #  fund+round fields to derived values
-    derived_values = {}
     application_id = application_json["id"]
-    print(f"deriving values for application id: {application_id}.")
     fund_round_shortname = "".join(application_json["reference"].split("-")[:2])
+    derived_values = {
+        "application_id": application_id,
+        "project_name": ""
+        if application_json["project_name"] is None and fund_round_shortname == "COFEOI"
+        else application_json["project_name"],
+        "short_id": application_json["reference"],
+        "fund_id": application_json["fund_id"],
+        "round_id": application_json["round_id"],
+        "funding_amount_requested": 0,
+        "asset_type": "No asset type specified.",
+        "language": application_json["language"],
+        "location_json_blob": {
+            "error": False,
+            "postcode": FIELD_DEFAULT_VALUE,
+            "county": FIELD_DEFAULT_VALUE,
+            "region": FIELD_DEFAULT_VALUE,
+            "country": FIELD_DEFAULT_VALUE,
+            "constituency": FIELD_DEFAULT_VALUE,
```

Comment:
Would prefer for this to be `null` and the frontend handle what to display. But OK if this is a lift and shift of the previous behaviour.

Reply:
Yeah it's just what was there before, didn't want to have to change the frontend as well